### ISSUE TITLE
First class tuples

### ISSUE THREAD

**Issue by 12345swordy** [dlang/projects#63]
write a dip that makes tuples built into the language itself to avoid writing tuple!(....) and other unneeded verbiage.

**Comment by UplinkCoder** [comment]
Do you mean tuple(), from std typecons?
or do you mean `AliasSeq` from std.meta?

In terms of aliasSeq that's something we actually need and I am working on it.

**Comment by 12345swordy** [comment]
tuple() from std typecons

**Comment by UplinkCoder** [comment]
we are a strongly typed language, why don't you create a struct instead?

**Comment by 12345swordy** [comment]
> we are a strongly typed language, why don't you create a struct instead?
That is a complete Non sequitur.  C# is a strongly typed language, yet it has first class tuples built in.

**Comment by andre2007** [comment]
There is a tuple dip started some years ago https://github.com/tgehr/DIPs/blob/tuple-syntax/DIPs/DIP1xxx-tg.md

**Comment by Geod24** [comment]
> we are a strongly typed language, why don't you create a struct instead?

There's a difference between duck typing and [structural typing](https://en.wikipedia.org/wiki/Structural_type_system).
That being said, there's a strong preference (or there was, from Andrei) to have things that can be built as library, be in a library, so this DIP is likely to be an uphill battle.

@12345swordy : Are you volunteering ?

**Comment by 12345swordy** [comment]
@Geod24 Yes I am willing to volunteer on this, if people are willing to assist me on this as I have time restraint on this.

**Comment by andre2007** [comment]
I wonder whether it makes sense to have it as a gsoc project: complete the DIP (https://github.com/tgehr/DIPs/blob/tuple-syntax/DIPs/DIP1xxx-tg.md)  and drive through the dip process?
Of course that means here is no coding at all to be written by the student...
